auth/realm/base/src/main/java/org/wildfly/security/auth/realm/CachingSecurityRealm.java

```diff
@@ -212,7 +212,22 @@ public boolean verifyEvidence(Evidence evidence) throws RealmUnavailableExceptio
                 if (evidence instanceof PasswordGuessEvidence) {
                     if (credentials.canVerify(evidence)) {
                         log.tracef("verifyEvidence For principal='%s' using cached credential", principal);
-                        return credentials.verify(evidence);
+                        boolean credentialsVerified = credentials.verify(evidence);
+                        if (!credentialsVerified) {
+                            log.tracef("verifyEvidence For principal='%s' using cached credential failed", principal);
+                            // since verification failed then verify evidence directly on an identity
+                            log.tracef("trying verifyEvidence For principal='%s' using underlying security realm", principal);
```

Comment:
As if both messages will always be logged if one is logged can they be combined into one?

Reply:
Updated the trace messages to be a single message. Thanks for review!